### PR TITLE
Change votingsetting client

### PR DESCRIPTION
**Pull request by alikhabazian**

### CODE REVIEW

File: scripts/src/client/changeVotingSetting.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    private async checkIsAnyRepetetive(newapprovers:string[]){
+        for (let approver of newapprovers) {
+            let isMember=await this.isMemberMultisig(approver)
+            if(isMember) throw new Error(`this new approver: ${approver} is already a member `);
+        }
+
+    }
+
+    private async sendProposal(proposalMetadata,daoActions:DaoAction[]){
+
```
Comment (mehrdadmms) [inline scripts/src/client/changeVotingSetting.ts:91]: One thing that you need to keep in mind is that the schema for create proposal is different between multisig plugin and token voting plugin.

**Review by mehrdadmms — COMMENTED** [review]
Great job!
This is only for multisig though, please also add token voting. 
User must be able to change token voting setting.
By the way please add the remove user from multisig as well. 

Overall the code is great and everything is in check. 

Continue with your work and let's test together when you're done :))

File: scripts/src/client/changeVotingSetting.ts [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+    async checkAreAllMember(formerapprovers:string[],votingAbi:any){
+        for (let approver of formerapprovers) {
+            let isMember=await this.isMember(approver,votingAbi)
+            if(!isMember) throw new Error(`this former approver: ${approver} is already not a member `);
+        }
+    }
+    async multisigRemoveAddresses(formerapprovers:string[]){
+        if (!this.initialized) throw new Error('This instance has not initialized yet, first call initial() method');
+        if (formerapprovers.length==0) throw new Error('There is not any formerapprover to remove');
+        await this.checkAreAllMember(formerapprovers,MULTISIGABI);
```
Comment (mehrdadmms) [inline scripts/src/client/changeVotingSetting.ts:214]:
```suggestion
        await this.checkAreAllMember(formerapprovers,MULTISIG_ABI);
```

File: scripts/src/client/changeVotingSetting.ts [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+        //porposal parent -> subdaoadminplugin -> childvoting(multisig)
+        const encodedFunctionDataMultisig=this.encodedFunctionData(MULTISIGABI,'removeAddresses',[formerapprovers])
```
Comment (mehrdadmms) [inline scripts/src/client/changeVotingSetting.ts:216]:
```suggestion
        const encodedFunctionDataMultisig=this.encodedFunctionData(MULTISIG_ABI,'removeAddresses',[formerapprovers])
```

File: scripts/src/client/changeVotingSetting.ts [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        const updateVotingSettingDaoAction: DaoAction[] = [
+            {
+                to: this.childVotingPluginContractAddress,
+                value: BigInt(0),
+                data: hexToBytes(encodedFunctionDataMultisig),
+            },
+        ];
+
+        const encodedFunctionDataSubDAOAdmin=this.encodedFunctionData(SUBDAOADMINABI,'execute',[updateVotingSettingDaoAction])
```
Comment (mehrdadmms) [inline scripts/src/client/changeVotingSetting.ts:225]:
```suggestion
        const encodedFunctionDataSubDAOAdmin=this.encodedFunctionData(SUB_DAO_ADMIN_ABI,'execute',[updateVotingSettingDaoAction])
```

**Review by mehrdadmms — COMMENTED** [review]
Awesome please also update readme.

File: scripts/src/client/changeVotingSetting.ts (hunk 1 above)
Comment (alikhabazian) [reply scripts/src/client/changeVotingSetting.ts:140]: If I understand correctly, you're emphasizing the importance of recognizing the distinction in the schema for creating proposals between the multisig plugin and the token voting plugin. I want to assure you that we have taken this into consideration, specifically acknowledging the necessity to be mindful of the parent DAO's voting type.

**Pull request merged**